src/ResourceManager/Network/Commands.Network/AzureFirewall/NewAzureFirewallCommand.cs

```diff
@@ -0,0 +1,147 @@
+    public class NewAzureFirewallCommand : AzureFirewallBaseCmdlet
+    {
+        private PSVirtualNetwork virtualNetwork;
+        private PSPublicIpAddress publicIpAddress;
+
+        [Alias("ResourceName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "location.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = "IpConfigurationParameterValues",
+            HelpMessage = "Virtual Network Name")]
+        [ValidateNotNullOrEmpty]
+        public string VirtualNetworkName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = "IpConfigurationParameterValues",
+            HelpMessage = "Public Ip Name")]
+        [ValidateNotNullOrEmpty]
+        public string PublicIpName { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "The list of AzureFirewallApplicationRuleCollections")]
+        public List<PSAzureFirewallApplicationRuleCollection> ApplicationRuleCollection { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "The list of AzureFirewallNetworkRuleCollections")]
+        public List<PSAzureFirewallNetworkRuleCollection> NetworkRuleCollection { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "A hashtable which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            var isVnetPresent = VirtualNetworkName==null;
+            
+            // Get the virtual network, get the public IP address
+            if (isVnetPresent)
```

Comment:
look at https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Billing/Commands.Billing/Cmdlets/BillingPeriods/GetAzureRmBillingPeriod.cs#L39for how to see which param set is used and using the appropriate parmas from the param set.

Reply:
thanks :) I believe I've fixed it now